spark/src/main/scala/org/apache/spark/sql/delta/commands/alterDeltaTableCommands.scala

```diff
@@ -775,6 +775,11 @@ case class AlterTableChangeColumnDeltaCommand(
       }
     }
 
+    if (originalField.dataType != newType) {
+      checkDependentExpressions(
+        spark, columnPath :+ columnName, txn.metadata, txn.protocol, "change")
```

Comment:
Does alter not fit better here because it comes out of the alter table ... alter column command?

Reply:
The syntax is `ALTER TABLE tableName (ALTER | CHANGE) COLUMN ..`, alter and change are both equally valid here